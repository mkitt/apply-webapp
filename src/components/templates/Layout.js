// @flow
import React, { type Node } from 'react'
import glamorous from 'glamorous'
import NavBar from '../../containers/NavBarContainer'

type Props = {
  children: Node,
}

export default (props: Props) => (
  <div>
    <NavBar />
    <Main role="main">
      {props.children}
    </Main>
  </div>
)

// -------------------------------------

const Main = glamorous.main({
  position: 'relative',
})
