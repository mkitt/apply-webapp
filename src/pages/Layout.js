// @flow
import React from 'react'
import glamorous from 'glamorous'
import NavBar from '../containers/NavBarContainer'

const Main = glamorous.main({
  position: 'relative',
})

type Props = {
  children: React.Element<*>,
}

export default (props: Props) => (
  <div>
    <NavBar />
    <Main role="main">
      {props.children}
    </Main>
  </div>
)
