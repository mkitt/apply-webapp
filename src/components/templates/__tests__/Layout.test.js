import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Layout from '../Layout'

describe('/login', () => {
  it('shows the correct snapshot tree for the Layout template', () => {
    const component = (
      <MemoryRouter>
        <Layout>
          <div>Children</div>
        </Layout>
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
