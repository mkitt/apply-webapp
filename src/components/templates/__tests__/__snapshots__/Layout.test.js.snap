// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/login shows the correct snapshot tree for the Layout template 1`] = `
.glamor-6,
[data-glamor-6] {
  height: 112px;
  background-color: #fff;
}

.glamor-5,
[data-glamor-5] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1392px;
  height: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-right: 8px;
  padding-left: 8px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  height: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-2,
[data-glamor-2] {
  font-size: 14px;
  color: #808080;
  margin-right: 40px;
}

.glamor-1,
[data-glamor-1] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

.glamor-7,
[data-glamor-7] {
  position: relative;
}

<div>
  <nav
    className="glamor-6"
  >
    <div
      className="glamor-5"
    >
      <div
        className="glamor-0"
      />
      <div
        className="glamor-0"
      >
        <span
          className="glamor-2"
        >
          <span>
            <span
              aria-label="hi"
              role="img"
            >
              ðŸ‘‹
            </span>
             Jeff Casimir
          </span>
          <span>
             / 
          </span>
          <a
            className="glamor-1"
            href="/login"
            onClick={[Function]}
          >
            Logout
          </a>
        </span>
        <a
          className="glamor-3"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="turing school mark"
            height={48}
            src="/turing-school-mark-256.png"
            width={48}
          />
        </a>
      </div>
    </div>
  </nav>
  <main
    className="glamor-7"
    role="main"
  >
    <div>
      Children
    </div>
  </main>
</div>
`;
